rotkehlchen/chain/ethereum/modules/makerdao/sai/decoder.py

```diff
@@ -0,0 +1,535 @@
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.asset == self.sai and
+                event.balance.amount == amount_paid
+            ):
+                event.event_subtype = HistoryEventSubType.PAYBACK_DEBT
+                event.counterparty = CPT_SAI
+                event.notes = f'Repay {amount_paid} {self.sai.symbol} to CDP {cdp_id}'
+                return None, []
+
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.PAYBACK_DEBT and
+                event.asset == self.sai and
+                event.balance.amount == amount_paid
+            ):
+                return None, []
+
+        if self.base_tools.is_tracked(depositor) is True:
+            event = HistoryBaseEntry(
+                event_identifier=transaction.tx_hash,
+                timestamp=ts_sec_to_ms(transaction.timestamp),
+                sequence_index=self.base_tools.get_sequence_index(tx_log),
+                location=Location.BLOCKCHAIN,
+                asset=self.sai,
+                event_type=HistoryEventType.SPEND,
+                event_subtype=HistoryEventSubType.PAYBACK_DEBT,
+                balance=Balance(amount=amount_paid),
+                location_label=depositor,
+                counterparty=CPT_SAI,
+                notes=f'Repay {amount_paid} {self.sai.symbol} to CDP {cdp_id}',
+            )
+            return event, []
+
+        return None, []
+
+    def _decode_liquidate_event(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],
+            action_items: Optional[list[ActionItem]],  # pylint: disable=unused-argument
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        """
+        This method decodes a liquidation event in a CDP
+
+        An example of such transaction is:
+        https://etherscan.io/tx/0x65d53653c584cde22e559cec4667a7278f75966360590b725d87055fb17552ba
+        """
+        liquidator = hex_or_bytes_to_address(tx_log.topics[1])
+        cdp_id = hex_or_bytes_to_int(tx_log.topics[2])
+
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.SPEND and
+                event.asset == self.peth and
+                event.counterparty == CPT_SAI
+            ):
+                return None, []
+
+        # check for the transfer event of the liquidation to Maker SaiTap contract
+        for log in all_logs:
+            if len(log.topics) < 2:
+                continue
+
+            if hex_or_bytes_to_address(log.topics[1]) == MAKERDAO_SAITUB_CONTRACT and hex_or_bytes_to_address(log.topics[2]) == MAKERDAO_SAITAP_CONTRACT:  # noqa: E501
```

Comment:
I don't understand what kind of even you are matching on here. You don't seem to check the topic[0] which is what defines the event,

Reply:
You're right, fixed!